_projects/2023/100516517/100516517.Rmd

```diff
@@ -0,0 +1,798 @@
+
+I organized the data by grouping it based on the 'Elections' and 'recat' columns. Then, I summarized the data by calculating the total number of deputies ('num_deputies' column) for each category and stored this information in a new column named 'total_deputies'.
+
+Two crucial adjustments were necessary to ensure the graph's accurate representation:
+
+1.  Transforming the 'recat' column into a factor specifying a certain order for its levels. This order was structured to mirror the original graph's arrangement: regional parties on the left, followed by a progressive ideological sequence from left-leaning parties to right-leaning formations.
+
+2.  Typically, graphs displaying an evolving trend have years or dates plotted along the X-axis. However, our graph's design was oriented differently with dates on the Y-axis. Attempting to flip the graph using 'coord_flip' resulted in an inverted orientation, placing the most recent years at the bottom. To resolve this, I converted the dates into decimal values, facilitating the reversal of their order. This adjustment ensured that the ticks representing election dates on the Y-axis were proportionally spaced according to the time between each election.
+
+```{r}
+#Creating the final graph database
+graph_base <- clean_data |> 
+  group_by(Elections, recat) |> 
+  summarise(total_deputies = sum(num_deputies)) |> 
+  ungroup() 
+
+#Ordering the political parties by their ideology
+graph_base$recat <- factor(graph_base$recat, levels = c("VOX", "PP", "Cs", "UCD", "CDS", "UPyD", "US", "PSOE", "SUMAR", "Más País", "PODEMOS", "PCE", "IU", "Otros catalanes", "CiU", "ERC", "Junts", "PNV", "HB", "EH Bildu", "CC", "Otros"))
+
+#Adding the DECIMAL_DATE column
+graph_base$DECIMAL_DATE <- decimal_date(graph_base$Elections)
+fechas_unicas <- unique(graph_base$DECIMAL_DATE)
+```
+
+# Replication: stream_graph
+
+The graph we aim to replicate is a stream graph. These visualizations demonstrate changes in the composition and distribution of multiple variables over time. They prove useful in depicting time series data and how the distribution of different categories or groups within these series evolves over time. In this instance, it's a proportional stream graph, meaning the width of the bands varies based on each category's proportion at a specific moment in time. Hence, although the number of deputies in the congress can fluctuate between 300 and 400, our graph will have a constant width, because it considers the proportion of deputies from each electoral party concerning the total number of deputies at any given time.
+
+The following code accomplishes the following:
+
+-   Creates the proportional flow graph using the "ggplot2" package.
+
+-   Flips the graph to make it vertical.
+
+-   Adjusts the axis limits to accommodate annotations.
+
+-   Establishes themes and styles, refining the appearance of the graph, including titles, subtitles, and axis formatting to mimic the original graph.
+
+```{r}
```

Comment:
Your graph is tall but narrow, and looks too small with the default height and width. Please add the `fig.asp` option and find the proper value so that the graph in the HTML looks nicer. Then apply this to all chunks.